numpy/__init__.pyi

```diff
@@ -633,7 +634,7 @@ test: PytestTester
 #
 # Placeholders for classes
 
-def show_config() -> None: ...
+def show_config(mode: ConfigDisplayModes) -> None | dict: ...
```

Comment:
@BvB93 / @mattip / @rgommers , just curious, how did the UT not catch this? I went over the typing UT but didn't quite get how it's not failing. Should we add some tests?

Reply:
> Should we add some tests?

The answer to that question is pretty much always "yes, if they are not slow and complicated"

Reply:
I'll take this up in a following PR. It's a bit hard to navigate the `typing` tests.